### PR TITLE
Add ServerJars Minecraft Auto-Updater

### PR DESCRIPTION
**Pull request by cixastra**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Did you branch your changes and PR from that branch and not from your master branch?

### New Server Submissions:

1. [x] Does your submission pass tests (server is connectable)?
2. [x] Does your server use a custom docker image?
   No, Server uses default Java Images (8,11,16, & 17)
3. [x] Have you added the server to the main README.md?
4. [x] Have you added a unique README.md for the server you are adding?

### CODE REVIEW

**Comment by cixastra** [comment]
bump^

**Review by gOOvER — CHANGES_REQUESTED** [review]
you missed to add it to Main Readmin in root

**Comment by cixastra** [comment]
The main readme has been changed? I don't quite understand what you mean.
Screen Shot 2022-02-05 at 6 32 28 PM

File: README.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ If you are reading this it looks like you are looking to add an egg to your serv
   * [Technic](game_eggs/minecraft/java/technic)
   * [Tuinity](game_eggs/minecraft/java/tuinity)
   * [VanillaCord](game_eggs/minecraft/java/vanillacord)
+  * [ServerJars AutoUpdater](game_eggs/minecraft/java/ServerJars)
```
Comment (parkervcp) [inline README.md:142]: This should be alphabetical

**Review by parkervcp — CHANGES_REQUESTED** [review]

**Comment by Software-Noob** [comment]
Hard-coding a specific version is a bad idea. Otherwise, this egg itself works fine, but I'm not a big fan of the idea itself.

**Comment by cixastra** [comment]
I understand the hesitation for using 3.0.0 as a hard coded version, however, traversing GitHub releases in ash is (1) difficult, and (2) not *really* necessary considering the software updates *very* slowly. It has been at 3.0.0 for a very long time now. As it works, Is this pr approved?

**Comment by Software-Noob** [comment]
> I understand the hesitation for using 3.0.0 as a hard coded version, however, traversing GitHub releases in ash is (1) difficult, and (2) not _really_ necessary considering the software updates _very_ slowly. It has been at 3.0.0 for a very long time now. As it works, Is this pr approved?

That's not a great philosophy when it comes back to bite us back in the ass. I'm against hard-coding specific releases. It's a very bad practice. We do have a [Github release grabber script.](https://github.com/parkervcp/eggs/blob/master/scripts/github_release_grabber.sh)

**Comment by cixastra** [comment]
I've implimented the release grabber, the script now grabs the latest jarfile. Test for success is complete and reveals no error.

File: game_eggs/minecraft/java/ServerJars/egg-server-jars.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "description": "The ServerJars Auto-Updater from https:\/\/serverjars.com",
+    "features": null,
+    "images": [
+        "ghcr.io\/pterodactyl\/yolks:java_8",
+        "ghcr.io\/pterodactyl\/yolks:java_11",
+        "ghcr.io\/pterodactyl\/yolks:java_16",
+        "ghcr.io\/pterodactyl\/yolks:java_17"
+    ],
+    "file_denylist": [],
+    "startup": "java -Xms{{SERVER_MEMORY}}M -Xmx{{SERVER_MEMORY}}M -Dterminal.jline=false -Dterminal.ansi=true -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https:\/\/mcflags.emc.gs\/ -Daikars.new.flags=true -Dfml.readTimeout=120 -Dfile.encoding=UTF-8 -jar {{SERVER_JARFILE}}",
```
Comment (Software-Noob) [inline game_eggs/minecraft/java/ServerJars/egg-server-jars.json:19]: Please keep the flags to a minimum. Most users have no idea what each one of them does in detail, which can cause issues when the server is behaving in unexpected ways regarding garbage collection or memory allocation.

File: game_eggs/minecraft/java/ServerJars/egg-server-jars.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
...\nif [ -z \"${GITHUB_USER}\" ] && [ -z \"${GITHUB_OAUTH_TOKEN}\" ] ; then\r\n    echo -e \"using anon api call\"\r\nelse\r\n    echo -e \"user and oauth token set\"\r\n    alias curl='curl -u ${GITHUB_USER}:${GITHUB_OAUTH_TOKEN} '\r\nfi\r\n\r\n## get release info and download links\r\nLATEST_JSON=$(curl --silent \"https:\/\/api.github.com\/repos\/${GITHUB_PACKAGE}\/releases\/latest\")\r\nRELEASES=$(curl --silent \"https:\/\/api.github.com\/repos\/${GITHUB_PACKAGE}\/releases\")\r\n\r\nif [ -z \"${VERSION}\" ] || [ \"${VERSION}\" == \"latest\" ]; then\r\n    DOWNLOAD_URL=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url | grep -i ${MATCH})\r\nelse\r\n    VERSION_CHECK=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .tag_name')\r\n    if [ \"${VERSION}\" == \"${VERSION_CHECK}\" ]; then\r\n        DOWNLOAD_URL=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .assets[].browser_download_url' | grep -i ${MATCH})\r\n    else\r\n        echo -e \"defaulting to latest release\"\r\n        DOWNLOAD_URL=$(echo ${LATEST_JSON} | jq .assets | jq -r .[].browser_download_url)\r\n    fi\r\nfi",
```
Comment (Software-Noob) [inline game_eggs/minecraft/java/ServerJars/egg-server-jars.json:28]: You copied the Github release grabber, but there is no  `VERSION` variable.

There should be an echo of some kind at the end to provide feedback when the installation is completed. Right now, there's nothing that might confuse someone.

**Review by Software-Noob — CHANGES_REQUESTED** [review]

**Pull request closed without merging**